src/index.ts

```diff
@@ -42,43 +42,51 @@ let tracer: ThundraTracer;
 module.exports = (options: any) => {
     const config = new ThundraConfig(options);
 
-    if (!(config.apiKey) || config.disableThundra) {
+    if (config.disableThundra) {
         return (originalFunc: any) => originalFunc;
     }
 
-    if (!(Utils.getConfiguration(envVariableKeys.THUNDRA_DISABLE_TRACE) === 'true') && config.traceConfig.enabled) {
-        const tracePlugin = TracePlugin(config.traceConfig);
-        config.plugins.push(tracePlugin);
-
-        tracer = tracePlugin.tracer;
-        config.metricConfig.tracer = tracer;
-        config.logConfig.tracer = tracer;
-        config.xrayConfig.tracer = tracer;
-        InvocationTraceSupport.tracer = tracer;
+    if (!(config.apiKey)) {
+        console.warn(`Thundra API Key is not given, instrumentation is disabled.`);
```

Comment:
I would use `Thundra API Key is not given, monitoring is disabled.` as error message as metric and logging is also disabled